java/vector/src/main/java/org/apache/arrow/vector/util/Text.java

```diff
@@ -202,15 +202,17 @@ public int find(String what, int start) {
   public void set(String string) {
     try {
       ByteBuffer bb = encode(string, true);
-      bytes = bb.array();
+      bytes = new byte[bb.remaining()];
+      bb.get(bytes);
+      bb.position(bb.position() - bytes.length); // move it back so we can decode it
```

Comment:
I reversed the change to check the original warning message, it doesn't show up anymore. 
This could be because of https://github.com/apache/arrow/pull/39452 and me not rebasing at the time :thinking: